### PR TITLE
Move logic of `sort_values` into the query compiler

### LINKED ISSUE #1577: Make sort_values implementation backend specific
Currently sort_vlaues is implemented using pandas sort_values + reindex. We need to make this method backend specific to implement different logic for OmniSci engine.

### PR DESCRIPTION
**Pull request by devin-petersohn**
Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1577 
- [x] tests ~added and~ passing

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -2791,6 +2791,23 @@ def sort_values(
...
+                ignore_index=ignore_index,
+            )
+        else:
+            result = self._query_compiler.sort_columns_by_row_values(
+                by,
+                ascending=ascending,
+                na_position=na_position,
+                ignore_index=ignore_index,
+            )
+        return self._create_or_update_from_compiler(result, inplace)
```
Comment (anmyachev) [inline modin/pandas/base.py:2809]: @devin-petersohn you need to remove code after `return` statement.

Comment (devin-petersohn) [reply modin/pandas/base.py:2809]: Thanks @anmyachev, I have removed it.

File: modin/pandas/base.py [hunk 2]
```diff
@@ -2791,59 +2791,22 @@ def sort_values(
...
-                na_position=na_position,
-            ).index
-            # Index without duplicates
-            new_index2 = broadcast_values2.sort_values(
-                by=by,
-                axis=axis,
+            result = self._query_compiler.sort_rows_by_column_values(
+                by,
                 ascending=ascending,
-                kind=kind,
```
Comment (anmyachev) [inline modin/pandas/base.py:2819]: It seems that you doesn't using `kind` parameter. Why so?

Comment (anmyachev) [reply modin/pandas/base.py:2819]: If you don't mind, let's delegate responsibility for processing `kind` parameter to query compilers.

File: modin/backends/pandas/query_compiler.py [hunk 3]
```diff
@@ -1860,3 +1860,78 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
...
+            The column or columns to sort by
+        ascending : bool
+            Sort in ascending order (True) or descending order (False)
+
+        Returns
+        -------
+        PandasQueryCompiler
+            A new query compiler that contains result of the sort
+        """
+        na_position = kwargs.get("na_position", "last")
```
Comment (anmyachev) [inline modin/backends/pandas/query_compiler.py:1879]:
```suggestion
        na_position = kwargs.get("na_position", "last")
        kind = kwargs.get("kind", "quicksort")
```

File: modin/backends/pandas/query_compiler.py [hunk 4]
```diff
@@ -1860,3 +1860,78 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
...
+            for col in columns
+        }
+        # Index may contain duplicates
+        broadcast_values1 = pandas.DataFrame(broadcast_value_dict, index=self.index)
+        # Index without duplicates
+        broadcast_values2 = pandas.DataFrame(broadcast_value_dict)
+        broadcast_values2 = broadcast_values2.reset_index(drop=True)
+        # Index may contain duplicates
+        new_index1 = broadcast_values1.sort_values(
+            by=columns, axis=0, ascending=ascending, na_position=na_position,
```
Comment (anmyachev) [inline modin/backends/pandas/query_compiler.py:1896]:
```suggestion
            by=columns, axis=0, ascending=ascending, kind=kind, na_position=na_position,
```

File: modin/backends/pandas/query_compiler.py [hunk 5]
```diff
@@ -1860,3 +1860,78 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
...
+        ).index
+        # Index without duplicates
+        new_index2 = broadcast_values2.sort_values(
+            by=columns, axis=0, ascending=ascending, na_position=na_position,
```
Comment (anmyachev) [inline modin/backends/pandas/query_compiler.py:1900]:
```suggestion
            by=columns, axis=0, ascending=ascending, kind=kind, na_position=na_position,
```

File: modin/backends/pandas/query_compiler.py [hunk 6]
```diff
@@ -1860,3 +1860,78 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
...
+        broadcast_value_list = [
+            self.getitem_row_array([row]).to_pandas() for row in rows
+        ]
+        index_builder = list(zip(broadcast_value_list, rows))
+        broadcast_values = pandas.concat(
+            [row for row, idx in index_builder], copy=False
+        )
+        broadcast_values.columns = self.columns
+        new_columns = broadcast_values.sort_values(
+            by=rows, axis=1, ascending=ascending, na_position=na_position,
```
Comment (anmyachev) [inline modin/backends/pandas/query_compiler.py:1935]:
```suggestion
            by=rows, axis=1, ascending=ascending, kind=kind, na_position=na_position,
```

File: modin/backends/pandas/query_compiler.py [hunk 7]
```diff
@@ -1860,3 +1860,78 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
...
+            The row or rows to sort by
+        ascending : bool
+            Sort in ascending order (True) or descending order (False)
+
+        Returns
+        -------
+        PandasQueryCompiler
+            A new query compiler that contains result of the sort
+        """
+        na_position = kwargs.get("na_position", "last")
```
Comment (anmyachev) [inline modin/backends/pandas/query_compiler.py:1922]:
```suggestion
        na_position = kwargs.get("na_position", "last")
        kind = kwargs.get("kind", "quicksort")
```

File: modin/backends/pandas/query_compiler.py (hunk 3 above)
Comment (devin-petersohn) [reply modin/backends/pandas/query_compiler.py:1879]: `kind` is not used in this implementation.

Comment (devin-petersohn) [reply modin/backends/pandas/query_compiler.py:1879]: I see all of your changes now, I will add them.

**Comment by anmyachev** [comment]
@YarShev are you OK with the last changes?

**Comment by YarShev** [comment]
@anmyachev , yes, LGTM!

**Pull request merged**